### DISCUSSION TITLE
User editing IFC

### DISCUSSION THREAD

**Discussion by NatMedrano** [IfcOpenShell/IfcOpenShell#4808] [General]
Hello, 

Does someone know if it is possible to the user draw in 2D and 3D using the IFC Open Shell? If so, how? If not, does someone know another solution for it, I mean something to integrate with the IFC Open Shell to accomplish the 2D and 3D drawings? 

ps: I am a beginner. 

Thank you for the help <3

**Comment by Moult** [comment]
Yes, this is how the BlenderBIM Add-on works - it uses IfcOpenShell to create drawings and geometry. See https://docs.ifcopenshell.org/ifcopenshell-python/geometry_creation.html as an example of how to get started.

**Comment by Vanuan** [comment]
I'd say the modeling UX in BlenderBIM is not easy. Creating IfcOpenShell Python scripts seems to be a much easier way to model.
Currently, BlenderBIM is a great IFC viewer. Even FreeCAD BIM has a much reacher interface. That being said, it's unfair comparison, as FreeCAD uses IfcOpenShell under the hood, so has a much narrower scope. FreeCAD 2D drawing module, on the other hand, is very poor. And BlenderBIM has a much higher level of customisation of drawings, even if uses InkScape as external helper tool.

To summarise:
- BlenderBIM has a strong 3D modeling visualisation compared to FreeCAD,
- FreeCAD has strong 3D modeling tooling, compared to BlenderBIM tools alone (excluding Blender capabilities).
- BlenderBIM has stronger 2D drafting workflow than FreeCAD's.
- FreeCAD benefits from IfcOpenShell data model while BlenderBIM developers are spread thin across the core library and the UI/UX.
- FreeCAD has a much stronger platform UI-wise (uses Qt), as BlenderBIM is built on top of quite limiting Blender UI (gpu, bgl and UILayout).
